### PR TITLE
[Feature] Allow for custom path to a language server binary

### PR DESCRIPTION
**Pull request by ichigo-gyuunyuu**
# Description

Users will now be able to use a different language server binary. The path can be set in `lv-confg.lua`, e.g. 
`O.lang.lua.lsp_path = "path/to/different/binary"`

Some languages like dart (already has `sdk_path`) and java (it's a special child :) ) have been left as is.

## How Has This Been Tested?

- Confirm that the langauge server attaches by checking `:LspInfo`
- Add a custom path in `lv-config.lua` and confirm that the change is reflected

### CODE REVIEW

**Review by Tastyep — APPROVED** [review]
Legit to me, we could maybe have done `lsp.path` instead of `lsp_path`, but since we don't have a use for other attributes, it's fine :+1:

**Comment by ichigo-gyuunyuu** [comment]
> Legit to me, we could maybe have done `lsp.path` instead of `lsp_path`, but since we don't have a use for other attributes, it's fine 👍

Yeah, that is more future-proof. Time for more refactoring if the lsp section gets more attributes 😃

**Comment by abzcoding** [comment]
> > Legit to me, we could maybe have done `lsp.path` instead of `lsp_path`, but since we don't have a use for other attributes, it's fine 👍
> 
> Yeah, that is more future-proof. Time for more refactoring if the lsp section gets more attributes 😃

I also agree with tasty, and would appreciate it if you use `lsp.path` instead of `lsp_path` 🥺

File: init.lua [hunk 1]
```diff
@@ -1,3 +1,4 @@
+print "hi"
```
Comment (abzcoding) [inline init.lua:1]: can you remove this line plz

Comment (ichigo-gyuunyuu) [reply init.lua]: I told myself to remove that before comitting 😞

**Review by abzcoding — APPROVED** [review]
tyvm

**Review by Tastyep — APPROVED** [review]

**Pull request merged**